[addons][screensaver] code improvement within CGUIWindowScreensaver

AlwinEsch opened the pull request:
## Description

There becomes the call amount of "CServiceBroker::GetWinSystem()->GetGfxContext()" reduced, with store of it as value and to reuse on following places.

This reduce within render flow the work about calls and on "CServiceBroker::GetWinSystem()->GetGfxContext()" to have an shorter line :smirk:.

Further done a whole clang cleanup on this file.

Also done this changes within my sandbox project and as unrelated to it, comes separate request here.

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## What is the effect on users?
<!--- Summarize the effect of this change on Kodi end-users. -->
<!--- If the PR does not have a noticeable impact (e.g., if it only changes documentation), -->
<!--- just leave it empty. Put in more detail the bigger the impact is. -->
<!--- This section may be used for automatic creation of release notes. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

AlwinEsch committed:
[addons][screensaver] clang cleanup of GUIWindowScreensaver.[cpp|h]

As on commit before an change now done here a "clang" cleanup to have in future correct on changes (and hopefully comes the time that's everything correct).

howie-f commented on xbmc/windows/GUIWindowScreensaver.cpp:
```diff
@@ -28,16 +28,20 @@ void CGUIWindowScreensaver::Process(unsigned int currentTime, CDirtyRegionList &
   MarkDirtyRegion();
   CGUIWindow::Process(currentTime, regions);
-  m_renderRegion.SetRect(0, 0, (float)CServiceBroker::GetWinSystem()->GetGfxContext().GetWidth(), (float)CServiceBroker::GetWinSystem()->GetGfxContext().GetHeight());
+  auto& context = CServiceBroker::GetWinSystem()->GetGfxContext();
```
`const auto&` :) please

AlwinEsch replied:
Have changed this line, as `GetWidth()` and `GetHeight()` defined as `const` it works. On the other two places not possible, as `ApplyStateBlock` and `CaptureStateBlock` do something.

howie-f reviewed (approved):
thanks, looks good to me

ksooo commented on xbmc/windows/GUIWindowScreensaver.cpp:
```diff
@@ -53,6 +57,8 @@ EVENT_RESULT CGUIWindowScreensaver::OnMouseEvent(const CPoint &point, const CMou
 
 bool CGUIWindowScreensaver::OnMessage(CGUIMessage& message)
 {
+  auto& context = CServiceBroker::GetWinSystem()->GetGfxContext();
```
Not sure I would unconditionally make this call here as you actually need the value only in two out of many `case`s... I would make the call only when actually needed, from within the `case`.

garbear replied:
For a regular variable I'd totally agree. However for a global I like to "break" least privilege as little as possible and as visibly as possible. See how I put the scope breakage at the beginning of the functions in [ProfileManager.cpp](https://github.com/xbmc/xbmc/blob/master/xbmc/profiles/ProfileManager.cpp#L377). Scope breakage happens all in one stanza there before everything else.

garbear reviewed (approved):
No issues that I can see, other than what @ksooo addressed.

AlwinEsch committed:
[addons][screensaver] code improvement within CGUIWindowScreensaver

There becomes the call amount of "CServiceBroker::GetWinSystem()->GetGfxContext()" reduced, with store of it as value and to reuse on following places.
This reduce within render flow the work about calls and on "CServiceBroker::GetWinSystem()->GetGfxContext()" to have an shorter line ;-).
